### PR TITLE
Mergify: configuration update

### PR DESCRIPTION
**Pull request by SteVwonder**
This change has been made by @SteVwonder from https://mergify.io config editor.

### CODE REVIEW

**Comment by SteVwonder** [comment]
Note: this depends on #26 being merged first

File: .mergify.yml
Comment (dongahn) [inline .mergify.yml:2]: Are we going to allow a merge without a rebase?

File: .mergify.yml
Comment (dongahn) [inline .mergify.yml:15]: Same question as above?

**Review by dongahn — COMMENTED** [review]
Just have a question about whether we will allow a merge with no rebase?

**Comment by dongahn** [comment]
Ok. Will wait until Issue #26 lands.

**Comment by dongahn** [comment]
BTW, great workflow automation and protection!

File: .mergify.yml
Comment (SteVwonder) [reply .mergify.yml:15]: A contributor having a merge commit in their PR is _technically_ allowed by the [current proposed contributing guide](https://github.com/ExaWorks/SDK/pull/25/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R30).  In that scenario, I see three options:

1. We fallback to merging (while still encouraging regular contributors to use the rebase workflow so that merging is rare)
3. We modify the contributing guide to require rebases/disallow merges and force the contributor to rebase their PR, dropping any merge commits (a daunting task for anyone unfamiliar with git and rebasing - so we may lose contributions)
2. We have the maintainers take on the responsibility of resolving conflicts during a rebase (seems like the worst option)

Either way, this seems like maybe a conversation to have over in #25, and then we codify the outcome of that here.  I'll make a comment in #25 referencing this.

**Comment by dongahn** [comment]
From the discussion at Issue #25, this is good to be merged? @SteVwonder ?

**Comment by SteVwonder** [comment]
Oops.  On a second look, it looks like I missed the `strict` merge option, which ensures that the CI goes green after updating the PR against the base branch (via rebase or merge) before merging into the base branch.  Let me add that quickly.

**Comment by SteVwonder** [comment]
Just rebased onto master. @dongahn, assuming this passes the CI, I believe this PR is ready to go in.

**Comment by dongahn** [comment]
Sounds good to me. I will add one "approval" which is required in our CI workflow as well.

**Review by dongahn — APPROVED** [review]

**Comment by SteVwonder** [comment]
Thanks! Since this updates the mergify config, someone will need to manually merge this, after which `merge-when-passing` should be enough.

**Pull request merged**
